### PR TITLE
[API] Add bundler dependency API endpoint

### PR DESCRIPTION
**Pull request by segiddins**
@indirect suggested I ought to do this

### CODE REVIEW

**Comment by indirect** [comment]
This looks right to me after perusing it. A smoke test for marshaled data and json data would be great, and then we'll need to add memcached to cache the per-gem results and librato stats so we still have Bundler usage and version stats.

File: app/controllers/api/v1/dependencies_controller.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      .where("rubygems.name IN (?) and indexed = true and (scope = 'runtime' or scope is null)", names)
+      .group(group_by_columns)
+      .order("rubygems.name, versions.created_at, number, platform, dep_name")
+      .pluck("#{group_by_columns}, #{dep_req_agg}, #{dep_name_agg}")
+  end
+
+  def dependencies_for(names)
+    dependencies_result(names).map do |r|
+      name, version, platform, checksum, ruby_version, created_at, dep_reqs, dep_names = r
+      rubygems_version = ">= 0" # TODO: get this from the DB
```
Comment (segiddins) [inline app/controllers/api/v1/dependencies_controller.rb]: this needs to be stored in the DB?

**Comment by segiddins** [comment]
@indirect is caching the per-gem results worth it, when it can mean making N queries instead of just 1?

**Comment by indirect** [comment]
Dammit, that was the wrong button. :(

**Comment by indirect** [comment]
@segiddins memcached supports multiget, so responses become a single memcached query rather than a complex (and slow) Postgres query. in real life experience, the api can keep up with the load no problem given per-gem memcached and cannot with pure queries.

File: app/jobs/indexer.rb [hunk 2]
```diff
@@ -82,13 +81,25 @@ def log(message)
...
-    # TODO: remove this after we upgrade rubygems client to 2.5.0+
-    @indexer ||=
-      begin
-        indexer = Gem::Indexer.new(Rails.root.join("server"), build_legacy: false)
-        def indexer.say(_) end
-        indexer
-      end
+  # TODO: remove this after we upgrade rubygems client to 2.5.0+
+  if Gem::VERSION < '2.5.0'
+    def self.indexer
```
Comment (arthurnn) [inline app/jobs/indexer.rb]: can you add a warning here to remind us of that?

**Pull request merged**